piplicenses/__init__.py

```diff
@@ -188,19 +188,29 @@ def parser(expression: str) -> set[str]:
     return parser
 
 
+def parse_licenses_list(licenses_str: str | None) -> list[str]:
+    if licenses_str is None:
+        return []
+
+    licenses = licenses_str.split(";")
+
+    # Strip items
+    licenses = [license.strip() for license in licenses]
```

Comment:
Do you set any benefits of your approach over the old one?

```suggestion
    licenses = map(str.strip, licenses)
```

Reply:
If it is about the conversion, then we might be able to remove the two proposed casts and do `return(licenses)`, removing one unnecessary assignment before the return as well. Thus I tend to prefer the functional programming approach of *map* and *filter* here, unless this has unexpected consequences with *mypy* and typing.

Reply:
# :+1: LGTM; With the caveat that the style can be improved as already suggested by @stefan6419846 

## TL;DR

I agree with @stefan6419846 on the use of map here, namely because it is a pre-defined function (e.g., `str.strip`) which will be slightly beter performance. and is just as readable. (For posterity the well discussed difference between list comprehension and map is typically about readability-performance trade-off, e.g. see [Geeks-for-geeks piece on it](https://www.geeksforgeeks.org/python/python-map-vs-list-comprehension/))

Additionally the only type impact is the list/iterator change and can be mitigated with `list()` as shown already.